pyro/infer/mcmc/hmc.py

```diff
@@ -144,10 +144,19 @@ def _iter_latent_nodes(trace):
                 yield (name, node)
 
     def _compute_trace_log_prob(self, model_trace):
+        # Wrap log_prob value in a tensor for JIT compatibility
+        if not model_trace.stochastic_nodes:
+            device = torch.zeros([]).device
+            for name, node in model_trace.nodes.items():
+                value = node.get("value", None)
+                if value is not None:
+                    device = value.device
+            return torch.zeros([], device=device)
         return self._trace_prob_evaluator.log_prob(model_trace)
 
     def _kinetic_energy(self, r):
-        r_flat = torch.cat([r[site_name].reshape(-1) for site_name in sorted(r)])
+        r_flat = torch.cat([r[site_name].reshape(-1) for site_name in sorted(r)]) if r \
+            else torch.tensor([])
```

Comment:
I'd recommend avoiding `torch.tensor([])` if you can; PyTorch does not have full support for empty tensors, i.e. many parts of the codebase appear to assume nonempty tensors. I would not be surprised if this was the line leading to your jit error.

Reply:
For example, can you return `None` instead?

Reply:
If we return None, we will need to short-circuit many parts of the code by checking that z is empty (or that KE is not None), which I wanted to avoid. But we can just return `0.` here.

That still does not resolve the JIT error when instead of `torch.zeros([])` we use `torch.tensor(0.)` anywhere in the model or the HMC code.